docs/readmes/proposals/p020_sgi_tunnel_transport.md

```diff
@@ -0,0 +1,114 @@
+---
+id: p020_sgi_tunnel_transport
+title: L3 transport for AGW
+hide_title: true
+---
+# L3 transport for SGi
+
+*Status: Draft*\
+*Authors: Pravin Shelar*\
+*Reviewers: Magma team*\
+*Last Updated: 2021-8-15*
+
+
+## **Objective**
+
+Add support for tunnel from AGW (SGi interface) to aggregator host.
+
+## **Motivation**
+
+Today AGW operator need to have separate networking device to create the tunneling infra for pushing SGi traffic to aggregator host. It also needs separate public IP which is expensive have these days. By providing support for tunneling traffic from AGW to aggregator host we eliminate both requirement and reduce operation cost of Magma AGW.
+This same tunnel can be potentially used for routing traffic to PGW aggregator host.
+
+## Design
+
+Initially IPSec tunnel was considered for tunneling traffic to the aggregator host. But after exploring other tunneling techniques it became clear that we need look at wireguard first. There are following advantages using wireguard over IPSec tunnels:
+
+1. wireguard performs better than IPsec or OpenVPN. There are bunch of benchmarking results on internet.
+2. It avoid dependency on OVS IPSec support.
+3. AGW can apply QoS policies before egressing traffic on the tunnel.
+
+So the plan is to implement wireguard based tunnel first and then implement IPSec. This doc covers both tunnel implementation.
+
+
+## **Detailed Design**
+
+This feature has two important aspects:
+
+1. Configuration of WireGuard tunnel
+2. Datapath programming
+
+### **Configuration:**
+
+AGW need following configuration for WireGuard tunnel:
+
+1. Install required binaries for WireGuard Tunnel
+2. Enable or disable SGi Tunnel
+3. Type of SGi Tunnel, wireguard, IPSec, etc
+4. Tunnel configuration:
+
+
+## **Datapath Programming for WireGuard Tunnel**
+
+![WireGuard datapath in NonNAT setup](assets/agw_wg_tun1.png)
+
+There are three parts in datapath programming.
+
+1. Setup the WG tunnel
```

Comment:
Is a `WG tunnel` a `WireGuard tunnel`? I think some explicit definitions here can be clarifying.

Reply:
yes, just removed all references of WG